src/test/java/com/redhat/cloud/notifications/IdentityTest.java

```diff
@@ -96,4 +96,37 @@ void testX509Identity() {
         Assertions.assertEquals("/CN=some-host.example.com",xid.getSubject());
 
     }
+
+    @Test
+    void testServiceAccountIdentity() {
+        String header =
+            "{\n" +
+                "\"identity\": {\n" +
+                "  \"org_id\"        : \"1234567\",\n" +
+                "  \"type\"            : \"ServiceAccount\",\n" +
+                "  \"service_account\" : {\n" +
+                "    \"username\"      : \"nonprod-fedramp-ocm\",\n" +
+                "    \"client_id\"     : \"xxx\",\n" +
+                "    \"scope\"      : \"api.notifications\"\n" +
+                "  },\n" +
+                "  \"internal\"       : {\n" +
+                "    \"org_id\"       : \"1234567\",\n" +
+                "    \"cross_access\" : false\n" +
+                "  }\n" +
+                "}" +
+            "}";
+
+        String xRhEncoded = null;
+        try {
+            xRhEncoded = new String(Base64.getEncoder().encode(header.getBytes("UTF-8")));
+        } catch (UnsupportedEncodingException e) {
+            Assertions.fail();
```

Comment:
This hides the exception in the test logs. We should add `throws Exception` to the test method instead.